extensions/control-plane/data-plane-transfer/src/main/java/org/eclipse/edc/connector/dataplane/transfer/DataPlaneTransferExtension.java

```diff
@@ -0,0 +1,163 @@
+import org.eclipse.edc.connector.dataplane.transfer.validation.ExpirationDateValidationRule;
+import org.eclipse.edc.connector.transfer.spi.callback.ControlPlaneApiUrl;
+import org.eclipse.edc.connector.transfer.spi.edr.EndpointDataReferenceTransformerRegistry;
+import org.eclipse.edc.connector.transfer.spi.flow.DataFlowManager;
+import org.eclipse.edc.jwt.TokenGenerationServiceImpl;
+import org.eclipse.edc.jwt.TokenValidationRulesRegistryImpl;
+import org.eclipse.edc.jwt.TokenValidationServiceImpl;
+import org.eclipse.edc.jwt.spi.TokenValidationService;
+import org.eclipse.edc.runtime.metamodel.annotation.Extension;
+import org.eclipse.edc.runtime.metamodel.annotation.Inject;
+import org.eclipse.edc.spi.message.RemoteMessageDispatcherRegistry;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.configuration.Config;
+import org.eclipse.edc.spi.types.TypeManager;
+import org.eclipse.edc.spi.types.domain.edr.EndpointDataReference;
+import org.eclipse.edc.web.spi.WebService;
+
+import java.time.Clock;
+
+import static java.lang.String.format;
+import static org.eclipse.edc.connector.dataplane.transfer.DataPlaneTransferConfig.DEFAULT_DPF_SELECTOR_STRATEGY;
+import static org.eclipse.edc.connector.dataplane.transfer.DataPlaneTransferConfig.DEFAULT_TOKEN_VALIDITY_SECONDS;
+import static org.eclipse.edc.connector.dataplane.transfer.DataPlaneTransferConfig.DPF_SELECTOR_STRATEGY;
+import static org.eclipse.edc.connector.dataplane.transfer.DataPlaneTransferConfig.TOKEN_VALIDITY_SECONDS;
+
+@Extension(value = DataPlaneTransferExtension.NAME)
+public class DataPlaneTransferExtension implements ServiceExtension {
+
+    public static final String NAME = "Data Plane Transfer";
+
+    /**
+     * This deprecation is used to permit a softer transition from the deprecated `web.http.validation` config group to
+     * the current `web.http.control`
+     *
+     * @deprecated "web.http.control" config should be used instead of "web.http.validation"
+     */
+    @Deprecated(since = "milestone8")
+    private static final String DEPRECATED_API_CONTEXT_ALIAS = "validation";
+
+    @Inject
+    private DataPlaneSelectorClient selectorClient;
+
+    @Inject
+    private ContractNegotiationStore contractNegotiationStore;
+
+    @Inject
+    private RemoteMessageDispatcherRegistry dispatcherRegistry;
+
+    @Inject
+    private WebService webService;
+
+    @Inject
+    private DataFlowManager dataFlowManager;
+
+    @Inject
+    private EndpointDataReferenceTransformerRegistry transformerRegistry;
+
+    @Inject
+    private Clock clock;
+
+    @Inject
+    private DataEncrypter dataEncrypter;
+
+    @Inject
+    private DataPlaneClient dataPlaneClient;
+
+    @Inject
+    private ControlApiConfiguration controlApiConfiguration;
+
+    @Inject
+    private KeyPairWrapper keyPairWrapper;
+
+    @Inject(required = false)
```

Comment:
Should this be mandatory? a Provider push dataflow controller without the control plane api url is a dataflow where the transfer can never succeed.

Reply:
Let's tackle it separately as PR is already pretty big ;)